### PR TITLE
Added null checks for incorrectly initialized AI ILogger.

### PR DESCRIPTION
**Pull request by dnduffy**

### CODE REVIEW

File: test/Microsoft.ApplicationInsights.AspNetCore.Tests/Logging/ApplicationInsightsLoggerTests.cs [hunk 1]
```diff
@@ -27,5 +27,15 @@ public void TestLoggerSetsSdkVersionOnLoggedTelemetry()
...
             Assert.True(isCorrectVersion);
         }
+
+        /// <summary>
+        /// Tests that an incorrectly constructed or uninitialized Application Insights ILogger does not throw exceptions.
+        /// </summary>
+        [Fact]
+        public void TestUninitializedLoggerDoesNotThrowExceptions()
+        {
+            ILogger logger = new ApplicationInsightsLogger("test", null, null);
```
Comment (Dmitry-Matveev) [inline test/Microsoft.ApplicationInsights.AspNetCore.Tests/Logging/ApplicationInsightsLoggerTests.cs]: Do we need to test out both telemetry client and filter being null separately?
